packages/graphql/src/graphql/directives/vector.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    DirectiveLocation,
+    GraphQLDirective,
+    GraphQLEnumType,
+    GraphQLInputObjectType,
+    GraphQLList,
+    GraphQLNonNull,
+    GraphQLString,
+} from "graphql";
+
+export const VectorProviderNames = ["VertexAI", "OpenAI", "AzureOpenAI", "Bedrock"] as const;
```

Comment:
May be unpopular, but... This is a GraphQL enum - should these be capitalized as is conventional?

```suggestion
export const VectorProviderNames = ["VERTEX_AI", "OPEN_AI", "AZURE_OPEN_AI", "BEDROCK"] as const;
```

Reply:
I would personally very much prefer to keep them as they are since they mirror the existing documentation and the configuration in the constructor

Reply:
I'm torn on this one, I'd say that we should match the enum standard in the rest of the API, but it is true that the constructor configuration should match this if possible :thinking: